### PR TITLE
feat:support for management-sdk and axios adapter

### PR DESCRIPTION
**Pull request by Amitkanswal**
changes:  support for management-sdk and axios adapter

### CODE REVIEW

File: src/types/common.types.ts [hunk 1]
```diff
@@ -12,16 +12,12 @@ type _HTTPMethods = 'DELETE' | 'GET' | 'HEAD' | 'PATCH' | 'POST' | 'PUT' | 'OPTI
 
 export type HTTPMethods = Uppercase<_HTTPMethods> | Lowercase<_HTTPMethods>
 
-export type ApiRequestProps = {
+export type RequestOption = {
         method: HTTPMethods,
-        url: string,
         headers?: GenericObjectType,
         body?: unknown,
```
Comment (MaitriGurey1) [inline src/types/common.types.ts:18]: will this still be body or data ? @Amitkanswal

File: src/uiLocation.ts [hunk 2]
```diff
@@ -157,10 +149,6 @@ class UiLocation {
             currentBranch: initializationData.currentBranch,
         });
 
-        this.api = (payload:ApiRequestProps)=> dispatchPostRobotRequest(postRobot, payload);
-
-        this.createAdapter = createSDKAdapter(postRobot);
```
Comment (MaitriGurey1) [inline src/uiLocation.ts:162]: why have we removed this ? @Amitkanswal

File: src/utils/adapter.ts [hunk 3]
```diff
@@ -1,45 +1,50 @@
...
-      .sendToParent("apiAdapter", opts)
-      .then(onData)
-      .then((data) => data)
-      .catch(onError);
-}
\ No newline at end of file
+export const createSDKAdapter = (postRobot: typeof PostRobot) => async (config: ApiRequestParams) => {
+  try {
+    if (config.data) {
+      if (typeof config.data === "string") {
```
Comment (rijil-tr) [inline src/utils/adapter.ts:16]: I don't think this is the solution we should be implementing. I think the problem is with forwarding the request to proxy service. 
if you decided to go with this solution then try something like this.
const body = typeof config.data === 'string' ? JSON.parse(config.data): config.data;

File: src/types/api.type.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export interface ApiResponse<T = any> {
+  data: T;
+  status: number;
+  statusText: string;
+  headers: Record<string, string>;
+  config: ApiRequestParams;
+  request?: any;
+}
+
+export type GenericAdapter = (config: ApiRequestParams) => Promise<ApiResponse>;
```
Comment (rijil-tr) [inline src/types/api.type.ts:21]: Where are we using this?

File: src/types/api.type.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  data?: unknown;
+  [key: string]: any;
+}
+
+export interface ApiResponse<T = any> {
+  data: T;
+  status: number;
+  statusText: string;
+  headers: Record<string, string>;
+  config: ApiRequestParams;
```
Comment (rijil-tr) [inline src/types/api.type.ts:17]: Why do we need config?

File: src/types/common.types.ts (hunk 1 above)
Comment (Amitkanswal) [reply src/types/common.types.ts:18]: it's body in app-sdk as we want to make it close to fetch. We are mapping it to data in AppExtensionComponent. To work with proxy service

File: src/uiLocation.ts (hunk 2 above)
Comment (Amitkanswal) [reply src/uiLocation.ts:162]: code is been refactor.  So position is been moved

File: src/utils/adapter.ts (hunk 3 above)
Comment (Amitkanswal) [reply src/utils/adapter.ts]: moving logic to AEC removing current code

File: src/types/api.type.ts (hunk 4 above)
Comment (Amitkanswal) [reply src/types/api.type.ts:21]: missed this removing it

File: src/types/api.type.ts (hunk 5 above)
Comment (Amitkanswal) [reply src/types/api.type.ts:17]: In response we are sending config back as response. Basically to debug. I can remove this. It's an additional info. If needed?

**Review by abhishek305 — APPROVED** [review]

File: __test__/uiLocation.test.ts [hunk 6]
```diff
@@ -142,12 +142,13 @@ describe("UI Location", () => {
 
         beforeEach(() => {
             mockPostRobot = postRobot;
-            opts = { method: 'GET', url: '/test' };
+            opts = { method: 'GET', url: '/test', baseURL: 'https://test.com' };
```
[the remark this answers is not in the document]
Comment (Amitkanswal) [reply __test__/uiLocation.test.ts]: done as discussed

File: src/types/api.type.ts (hunk 5 above)
Comment (Amitkanswal) [reply src/types/api.type.ts]: removed

**Pull request merged**